compiler/src/iree/compiler/Codegen/Dialect/GPU/Transforms/Transforms.cpp

```diff
@@ -19,6 +19,62 @@
 
 namespace mlir::iree_compiler::IREE::GPU {
 
+//===----------------------------------------------------------------------===//
+// MultiMmaOp Unit Dim Folding
+//===----------------------------------------------------------------------===//
+
+namespace {
+struct DropMultiMmaUnitDimsPattern
+    : public OpRewritePattern<IREE::GPU::MultiMmaOp> {
+  using OpRewritePattern<IREE::GPU::MultiMmaOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(IREE::GPU::MultiMmaOp mmaOp,
+                                PatternRewriter &rewriter) const override {
+    if (mmaOp.hasTensorSemantics()) {
+      return rewriter.notifyMatchFailure(
+          mmaOp, "unimplemented: unit dim dropping for tensor mma ops");
+    }
+    SmallVector<int64_t> bounds;
+    mmaOp.getIterationBounds(bounds);
+    if (bounds.empty()) {
+      return rewriter.notifyMatchFailure(mmaOp, "no dimensions to fold");
+    }
+
+    if (!llvm::all_of(bounds, [](int64_t b) { return b == 1; })) {
+      return rewriter.notifyMatchFailure(mmaOp,
+                                         "not all iteration bounds are unit");
+    }
```

Comment:
Why does this need to be true? Can you add a comment explaining why we have this?